### PR TITLE
FOGL-5402.test fixes

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 1]
```diff
@@ -255,40 +171,22 @@ async def test_bad_delete_cert(self, client, cert_name, actual_code, actual_reas
         assert {"message": actual_reason} == json_response
 
     async def test_bad_delete_cert_if_in_use(self, client):
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:173]: test_bad? I think we should keep `test_delete_cert_if_configured_to_use`

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 2]
```diff
@@ -300,24 +198,18 @@ async def test_bad_type_delete_cert(self, client):
             assert 1 == patch_cfg.call_count
 
     @pytest.mark.parametrize("cert_name, param", [
-        ('fledge.cert', '?type=cert'),
         ('fledge.json', '?type=cert'),
         ('fledge.pem', '?type=cert'),
-        ('fledge.key', '?type=key'),
         ('test.cer', '?type=cert'),
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:203]: would it not it better to keep .key .cert check as well? may with with name foo.key foo.cert

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 3]
```diff
@@ -352,6 +239,195 @@ async def test_delete_cert(self, client, certs_path, cert_name='server.cert'):
                             assert '{} has been deleted successfully'.format(cert_name) == json_response['result']
                         assert 1 == patch_remove.call_count
 
+    async def test_upload(self, client, certs_path):
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:242]: test_upload_default_installed

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 4]
```diff
@@ -352,6 +239,195 @@ async def test_delete_cert(self, client, certs_path, cert_name='server.cert'):
...
+        patch_logger_info = mocker.patch.object(middleware._logger, 'info')
+        patch_validate_token = mocker.patch.object(User.Objects, 'validate_token', return_value=_rv1)
+        patch_refresh_token = mocker.patch.object(User.Objects, 'refresh_token_expiry', return_value=_rv2)
+        patch_user_get = mocker.patch.object(User.Objects, 'get', return_value=_rv3)
+        return patch_logger_info, patch_validate_token, patch_refresh_token, patch_user_get
+
+    async def test_bad_upload_when_admin_role_is_required(self, client, certs_path, mocker):
+        files = {'key': open(str(certs_path / 'certs/fledge.key'), 'rb'),
+                 'cert': open(str(certs_path / 'certs/fledge.cert'), 'rb')}
+        patch_logger_info, patch_validate_token, patch_refresh_token, patch_user_get = await self.auth_token_fixture(
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:290]: very nice!

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 5]
```diff
@@ -352,6 +239,195 @@ async def test_delete_cert(self, client, certs_path, cert_name='server.cert'):
...
+        patch_logger_info.assert_called_once_with('Received %s request for %s', 'POST', '/fledge/certificate')
+
+    async def test_bad_upload_when_cert_in_use_and_with_non_admin_role(self, client, certs_path, mocker):
+        files = {'cert': open(str(certs_path / 'certs/test.cer'), 'rb')}
+        patch_logger_info, patch_validate_token, patch_refresh_token, patch_user_get = await self.auth_token_fixture(
+            mocker, is_admin=False)
+        msg = 'Certificate with name test.cer is configured to be used, ' \
+              'An `admin` role permissions required to add/overwrite.'
+        # Changed in version 3.8: patch() now returns an AsyncMock if the target is an async function.
+        cat_info = {'certificateName':  {'value': 'test'},  'authCertificateName':  {'value': 'foo'}}
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:313]: we better parametrize with test.cert and foo.cert; or patch cat info 
> cat_info = {'certificateName':  {'value': 'test'},  'authCertificateName':  {'value': 'ca'}}

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 6]
```diff
@@ -352,6 +239,195 @@ async def test_delete_cert(self, client, certs_path, cert_name='server.cert'):
...
+                    result = await resp.text()
+                    json_response = json.loads(result)
+                    assert {"message": msg} == json_response
+        patch_logger.assert_called_once_with(msg)
+        patch_user_get.assert_called_once_with(uid=2)
+        patch_refresh_token.assert_called_once_with(ADMIN_USER_HEADER['Authorization'])
+        patch_validate_token.assert_called_once_with(ADMIN_USER_HEADER['Authorization'])
+        patch_logger_info.assert_called_once_with('Received %s request for %s', 'POST', '/fledge/certificate')
+
+    async def test_upload(self, client, certs_path, mocker):
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:333]: test_upload_as_admin

**Review by praveen-garg — COMMENTED** [review]
looks great! Few comments to consider.

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py (hunk 2 above)
Comment (ashish-jabble) [reply tests/unit/python/fledge/services/core/api/test_certificate_store.py:203]: In that case we need to add new test data files

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**